woocommerce/woocommerce
# Add notice about webhooks using the legacy REST API

### Changes proposed in this Pull Request:

This pull request is an extension of https://github.com/woocommerce/woocommerce/pull/40535. It:

1. Adds a new `WebhookUtil::get_legacy_webhooks_count` method. It counts the existing webhooks that are configured to use the Legacy REST API classes to generate their payloads.
2. Adds a new `api_version` optional argument to ` WC_Webhook_Data_Store::search_webhooks`.
3. Adds a new "Legacy" section to the webhooks settings page, it lists the same webhooks that are counted by `get_legacy_webhooks_count`.
4. Adds a new warning notice that is shown if at least one of these legacy webhooks exist and the separate Legacy REST API extension doesn't exist:

![image](https://github.com/woocommerce/woocommerce/assets/937723/7a16bd05-6c1a-4ebd-a4c8-be564adacfe2)

Note that if legacy webhooks exist _and_ the Legacy REST API is enabled, two notices will appear. This is on purpose, it's better to be "overinformative" in this case and the notices can be dismissed anyway.

### How to test the changes in this Pull Request:

1. If you tested https://github.com/woocommerce/woocommerce/pull/40535 and have the dummy Legacy REST API plugin installed, disable it.
2. Make sure that no webhooks exist that are configured with "Legacy API v3" in the "API Version" field (WooCommerce - Settings - Advanced - Webhooks, you need to edit a webhook to see the API version it's using).
3. Run the following to simulate a WooCommerce update from the point of view of the admin notices: `wp eval 'WC_Admin_Notices::reset_admin_notices();'`
4. Verify that no new notices appear when reloading any admin page.
5. Create a couple of new webhooks (or edit existing ones) so that they have "Legacy API v3" in the "API Version" field.
6. Repeat step 3.
7. Reload the webhooks list page. Note how the "The WooCommerce webhooks with Legacy REST API payload will be unsupported soon" notice has appeared and that there's a new "Legacy" section.
8. Go to the new "Legacy" section, verify that it lists the appropriate webhooks.
8. Delete the webhooks, or edit them to use a different API version.
9. Reload any admin page, the notice should have disappeared.
10. Repeat step 5, then step 3, the notice should appear again upon reload.
11. Create a dummy Legacy REST API plugin (see the testing instructions of https://github.com/woocommerce/woocommerce/pull/40535, step 11) unless you have done that already. Activate the plugin.
12. Repeat step 3, reload, the notice should have disappeared.
13. Try also manually dismissing the notice (to revert that: `wp db query "delete from wp_usermeta where meta_key='dismissed_legacy_webhooks_unsupported_in_woo_90'"`).

### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
